crates/typst-library/src/math/accent.rs

```diff
@@ -188,7 +188,7 @@ cast! {
     self => self.0.into_value(),
     v: char => Self::new(v),
     v: Content => match v.to_packed::<SymbolElem>() {
-        Some(elem) => Self::new(elem.text),
-        None => bail!("expected a symbol"),
+        Some(elem) if elem.text.chars().count() == 1 => Self::new(elem.text.chars().next().unwrap()),
+        _ => bail!("expected a single-codepoint symbol"),
```

Comment:
this could use the same `parse` approach as in `Delimiter`, right? I didn't know about `FromStr for char`, that's neat!